pkg/kudoctl/util/github/client.go

```diff
@@ -84,19 +85,19 @@ func NewGithubClient(cred string) (*GithubClient, error) {
 		tp.OTP = cred
 	}
 
-	return &GithubClient{client: client}, nil
+	return &Client{client: client}, nil
 }
 
 // GetMostRecentFrameworkContentDir returns the content of the most recent Framework
-func (g *GithubClient) GetMostRecentFrameworkContentDir(framework string) (*github.RepositoryContent, error) {
```

Comment:
Naming and consistency is one of the hardest things and the earlier we can enforce it and set good examples the merrier.

With that said, I agree on this change and everything that follows this overarching goal of consistency.

For the sake of it, allow me to explain my reasoning in the first place:
It was inspired by the Kubernetes project using e.g. `func (c *nodes)` ( ref https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/node.go#L132 or https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/extensions/v1beta1/replicaset.go#L70 ) which seemed uncommon `in Go when making methods for the variable name to match the first letter of the struct the method is implemented on`, so I was following here just upstream.

Alternatives are also imo `gc` for `GithubClient` so it makes it easier in the code when reading e.g. `c.client.Repositories.GetContents`. For the UX it would mean in my opinion that it is not immediate obvious to the reader what client is meant with it. E.g. `gc` defined in `gc *github.Client` is cleaner to me then. And we do it in other parts such https://github.com/kudobuilder/kudo/pull/182/files#diff-47795ca62796b678e9c01915094311ddR86 or `k2c` ( `k2c *kudo.Client` ), which means also if changing we should stay consistent in those other parts as well. Certainly something I want to caution to keep in the back of our minds.

Reply:
:+1: Understand that reasoning. I trend towards more of the Go Style Guide versus the Kubernetes repository which traditionally doesn't necessarily follow best practices. Our repository will start to look more like good Go.

https://github.com/golang/go/wiki/CodeReviewComments#receiver-names
https://golang.org/doc/effective_go.html

The method receiver should still be `c` because the type is Client (which happens to be in the github package). But yeah, we should stay consistent.

Let's separately come up with a style guide we want to enforce. Those docs are two good canonical sources to start with.

Reply:
More on this:

https://peter.bourgon.org/go-for-industrial-programming/
https://peter.bourgon.org/blog/2017/06/09/theory-of-modern-go.html
https://peter.bourgon.org/go-best-practices-2016/

Reply:
This one's important too:

https://talks.golang.org/2014/names

Reply:
we on the same page in our desires... a standard and consistency.   I do like @fabianbaier reason for `gc *github.Client` but fall inline with general consensus and consistency.  +1 on the **style guide** .